pkg/kubelet/util/swap/swap_util.go

```diff
@@ -0,0 +1,125 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package swap
+
+import (
+	"bytes"
+	"fmt"
+	"k8s.io/klog/v2"
+	"k8s.io/mount-utils"
+	"os"
+	sysruntime "runtime"
+	"strings"
+	"sync"
+)
+
+var (
+	tmpfsNoswapOptionSupported        bool
+	tmpfsNoswapOptionAvailabilityOnce sync.Once
+	swapOn                            bool
+	swapOnErr                         error
+	swapOnOnce                        sync.Once
+	rootDir                           string
+)
+
+const TmpfsNoswapOption = "noswap"
+
+// InitializeRootDir gets a path for a base directory, to which we have write permission.
+// This path will be later use for writing temporary files / mount points / etc.
+func InitializeRootDir(path string) {
+	rootDir = path
+}
+
+func IsTmpfsNoswapOptionSupported(mounter mount.Interface) bool {
```

Comment:
There is a potential optimisation you can do here to avoid having to do the mount test.

You can check if the kernel is > 6.4, and if so, short-circuit and return `true` since the feature will most definitely be there. But then, for anything else, you can do the mount attempt to verify things 100% (and support older kernels with backported changes).

We already do this to discover whether some kernel features are supported, per:

- [pkg/util/kernel/version.go](https://github.com/kubernetes/kubernetes/blob/master/pkg/util/kernel/version.go)
- [pkg/util/kernel/constants.go](https://github.com/kubernetes/kubernetes/blob/master/pkg/util/kernel/constants.go)
- [pkg/proxy/ipvs/proxier.go#L299-L314](https://github.com/kubernetes/kubernetes/blob/master/pkg/proxy/ipvs/proxier.go#L299-L314)

Thoughts?

Reply:
Awesome! Thanks a lot, I've added that